src/accelerate/accelerator.py

```diff
@@ -1613,6 +1619,11 @@ def save(self, obj, f):
         """
         save(obj, f)
 
+    def register_save_state_pre_hook(self, hook: Callable[..., None]) -> hooks.RemovableHandle:
+        handle = hooks.RemovableHandle(self._save_model_state_pre_hook)
+        self._save_model_state_pre_hook[handle.id] = hook
+        return handle
+
```

Comment:
Can we combine `register_save_state_pre_hook` and `register_load_state_pre_hook` using a bool parameter `load_state`

```suggestion
    def register_pre_hook(self, hook: Callable[..., None], load_state: bool) -> hooks.RemovableHandle:
        if load_state:
            pre_hook = self._load_model_state_pre_hook
        else:
            pre_hook = self._save_model_state_pre_hook
        handle = hooks.RemovableHandle(pre_hook)
        pre_hook[handle.id] = hook
        return handle

```

Reply:
@pacman100 WDYT of instead mimicking how we have `prepare` and have a `register_state_pre_hook` func that calls internally `register_save_state_pre_hook` or `register_load_state_pre_hook`? I also think the explicit naming on `save` vs `load` pre-hook is important here for users to understand what it does, but have `save_state_pre_hook` for convenience
potentially.